src/bin/add/main.rs

```diff
@@ -143,6 +147,34 @@ fn handle_add(args: &Args) -> Result<()> {
     }
     let deps = &args.parse_dependencies()?;
 
+    if let Some(ref requested_features) = args.features {
+        let available_features = deps
+            .iter()
+            .map(|d| d.available_features.clone())
+            .flatten()
+            .collect::<Vec<String>>();
+        let flattened_features: Vec<String> = requested_features
+            .iter()
+            .map(|feat| {
+                feat.split(' ')
+                    .map(|s| s.trim().to_string())
+                    .collect::<Vec<String>>()
+            })
+            .flatten()
+            .collect();
+        let unknown_features: Vec<String> = flattened_features
+            .iter()
+            .filter(|req_feat| !available_features.contains(req_feat) && !req_feat.is_empty())
+            .cloned()
+            .collect();
+
+        if !unknown_features.is_empty() {
+            return Err(
+                ErrorKind::NoSuchFeaturesFound(unknown_features, available_features).into(),
+            );
```

Comment:
Can we start this off as a warning instead of an error?  I'm not confident we are doing everything to pull all of the features (e.g.  optional dependencies can act as features). 

This way users can verify and move on, rather than get blocked due to any bugs in our logic.  As more people test and find holes, we can look at turning it into a hard error.

Reply:
fair, I'll make the change